mico-admin/src/app/service-detail-status/service-detail-status.component.ts

```diff
@@ -1,18 +1,39 @@
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, OnInit, Input, OnDestroy } from '@angular/core';
+import { ApiService } from '../api/api.service';
+import { Subscription } from 'rxjs';
 
 @Component({
     selector: 'mico-service-detail-status',
     templateUrl: './service-detail-status.component.html',
     styleUrls: ['./service-detail-status.component.css']
 })
-export class ServiceDetailStatusComponent implements OnInit {
+export class ServiceDetailStatusComponent implements OnInit, OnDestroy {
 
-    @Input() serviceStatus;
+    @Input() shortName;
+    @Input() version;
 
+    subServiceStatus: Subscription;
 
-    constructor() { }
+    serviceStatus;
+
+    constructor(
+        private apiService: ApiService
+    ) { }
 
     ngOnInit() {
+        if (this.shortName == null || this.version == null) {
+            return;
+        }
+
+        this.apiService.getServiceStatus(this.shortName, this.version).subscribe(val => {
+            this.serviceStatus = JSON.parse(JSON.stringify(val));
```

Comment:
is there a need too copy this value?

Reply:
not necessary. However, at some point we want to be able to change deployment details like the number of replicas. This could be done at this view analog to the modification of application/service attributes. For this (future) feature a deep copy would be needed.